src/qt/bitcoingui.cpp

```diff
@@ -982,12 +1001,15 @@ void BitcoinGUI::showEvent(QShowEvent *event)
 }
 
 #ifdef ENABLE_WALLET
-void BitcoinGUI::incomingTransaction(const QString& date, int unit, const CAmount& amount, const QString& type, const QString& address, const QString& label)
+void BitcoinGUI::incomingTransaction(const QString& date, int unit, const CAmount& amount, const QString& type, const QString& address, const QString& label, const QString& walletName)
 {
     // On new transaction, make an info balloon
     QString msg = tr("Date: %1\n").arg(date) +
-                  tr("Amount: %1\n").arg(BitcoinUnits::formatWithUnit(unit, amount, true)) +
-                  tr("Type: %1\n").arg(type);
+                  tr("Amount: %1\n").arg(BitcoinUnits::formatWithUnit(unit, amount, true));
+    if (WalletModel::isMultiwallet() && !walletName.isEmpty())  {
+        msg += tr("Wallet: %1\n").arg(walletName);
```

Comment:
Seems to need a line break above:
<img width="362" alt="wallet" src="https://user-images.githubusercontent.com/10217/37042990-94fbbe50-212d-11e8-89f5-0ebc1abba96e.png">

Reply:
It's fixed right? At least there's `\n` in the `Amount: ` above.